R/checks.R

```diff
@@ -207,6 +207,32 @@ check_form4.1_box5 <- function(form) {
                         "Textfield and coding box 5 of form4.1 is completed if and only if the other option is chosen"))
 }
 
+#' Checks that if antibiotics was first administered at the injury scene,
+#' time from injury to antibiotics administration is at most 5 hours before time from injury to hospital admission
+#' 
+#' @param form a form containing ptstatus and form 4.1 
+#' 
+#' @return a data frame containing all the invalid rows
+#' 
+#' @import tidyverse
+#' @export
+check_abx_locabx_1 <- function(form){
+  abx_to_admission_difference_cutoff <- 5
+  problems <- form %>%
+    transmute(
+      region, site, studyid, ptinit, ptstatus, abx, locabx, iaunits, iahrs, iadays, 
+      ihunits, ihhrs, ihdays,
+      comment = "For antibiotics first administered at the injury scene, time from injury to antibiotics administration should be at most 5 hours before time from injury to hospital admission") %>%
+    filter(ptstatus == 1 & locabx == 1 & abx == 1 &
+             ((iaunits == 1 & 
```

Comment:
consider using if_else for better readability and performance.

Reply:
I think I need better clarification:
if () {} else {} is the normal if else we usually use.
Tidyverse provides if_else() which is a vectorized version,
`if_else(iaunits == 1,
if_else(ihunits == 1,
ihhrs - iahrs > abx_to_admission_difference_cutoff , ihdays * 24 - iahrs > abx_to_admission_difference_cutoff)
if_else(ihunits == 2, 
ihhrs - iadays * 24 > abx_to_admission_difference_cutoff,
ihdays * 24 - iadays * 24 > abx_to_admission_difference_cutoff )`